utils/build/docker/python/flask/app.py

```diff
@@ -166,6 +171,71 @@ def dbm():
     return Response(f"Integration is not supported: {integration}", 406)
 
 
+@app.route("/dsm")
+def dsm():
+    topic = "dsm-system-tests-queue"
+    consumer_group = "testgroup1"
+    print(os.environ)
+
+    def delivery_report(err, msg):
+        if err is not None:
+            print(f"[kafka] Message delivery failed: {err}")
+        else:
+            print(f"[kafka] Message delivered to {msg.topic()} [{msg.partition()}]")
+
+    def produce():
+        producer = Producer({
+            'bootstrap.servers': 'kafka:9092',
+            'client.id': "python-producer"
+        })
+        message = b"Hello, Kafka!"
+        producer.produce(topic, value=message, callback=delivery_report)
+        producer.flush()
+        print("[kafka] Produced and flushed message")
+
+    def consume():
+        consumer = Consumer(
+            {
+                'bootstrap.servers': 'kafka:9092',
+                'group.id': consumer_group,
+                'enable.auto.commit': True,
+                'auto.offset.reset': 'earliest',
+            })
+
+        consumer.subscribe([topic])
+
+        msg_received = False
+        while not msg_received:
+            msg = consumer.poll(10)
+            if msg is None:
+                print("[kafka] Consumed message but got nothing")
+            elif msg.error():
+                print("[kafka] Consumed message but got error " + msg.error())
+            else:
+                print("[kafka] Consumed message: ")
+                print(f"[kafka] from topic {msg.topic()}, partition {msg.partition()}, offset {msg.offset()}, key {str(msg.key())}")
+                print(f"[kafka] value {msg.value()}")
+                msg_received = True
+        consumer.close()
+
+    integration = flask_request.args.get("integration")
+    print(f"[kafka] Got request with integration: {integration}")
+    if integration == "kafka":
+        print("DD_DATA_STREAMS_ENABLED")
+        print(os.environ['DD_DATA_STREAMS_ENABLED'])
+        print(os.getenv('DD_DATA_STREAMS_ENABLED'))
+        produce_thread = threading.Thread(target=produce, args=())
+        consume_thread = threading.Thread(target=consume, args=())
+        produce_thread.start()
+        consume_thread.start()
+        produce_thread.join()
+        consume_thread.join()
+        print("[kafka] returning response")
+        return Response("ok")
```

Comment:
this gets printed, but in `test_dsm.py` the response object is mostly empty, and self.r.Text is None